docs/web/howto/web-in-60/websockets.rst

```diff
@@ -0,0 +1,91 @@
+
+WebSocket Server
+----------------
+
+Let's build a demo of a simple websocket server communicating with a web browser.
+To begin with, we will need a folder with 3 files in it.
+First, let's do the Twisted part.  We need a :py:class:`twisted.web.websocket.WebSocketResource` to be served at a known URL, so let's put one into a ``.rpy`` file called ``webskt.rpy`` :
+
+:download:`webskt.rpy <../../examples/websocket/webskt.rpy>`
+
+.. literalinclude:: ../../examples/websocket/webskt.rpy
+
+Note that ``WebSocketEchoFactory`` complies with the :py:class:`twisted.web.websocket.WebSocketServerFactory` :py:class:`protocol <typing.Protocol>`, returning a ``WebSocketDemo`` that complies with :py:class:`twisted.web.websocket.WebSocketProtocol` .
+We implement ``negotiationFinished``, the method called once the websocket connection is fully set up, to begin sending a text message to our peer once per second.
+
+Then, we will need an index page for our live websocket site, with a button on it that hooks up a JavaScript function to connect to ``/webskt.rpy``:
+
+:download:`index.html <../../examples/websocket/index.html>`
+
+.. literalinclude:: ../../examples/websocket/index.html
+
+Finally, we need our JavaScript source code that actually does the connecting of various events.
+Learning how to program in JavaScript is a bit outside the scope of this tutorial, but hopefully its function is obvious - we connect a websocket to the URL ``ws://localhost:8080/webskt.rpy`` , and then hook up event-handlers for when the socket connects, receives messages, receives errors, and closes:
+
+:download:`script.js <../../examples/websocket/script.js>`
+
+.. literalinclude:: ../../examples/websocket/script.js
+
+Note that, upon connect, the web socket sends a message to the server.
+
+If you were to put these all into a folder called ``my-websocket-demo`` , you can use ``twist`` to serve them, via ``twist web --path ./my-websocket-demo``.
+Then, you can open up a web browser, pointed at ``http://localhost:8080/`` and see a "connect websocket" button.
+Click the button, and you should see the web page populate with text like this:
+
+.. code-block::
+
+   socket opened: «{"isTrusted":true}»
+   message received: «heartbeat 1»
+   message received: «reply to hello world»
+   message received: «heartbeat 2»
+   message received: «heartbeat 3»
+   message received: «heartbeat 4»
+
+And that's all you need to implement a websocket server with Twisted!
+
+Since :py:class:`twisted.web.websocket.WebSocketResource` is a standard Twisted :py:class:`resource <twisted.web.server.Resource>`, you can integrate it with things like :doc:`authentication <http-auth>` or :doc:`sessions <session-basics>`, just as you would any other resource.
+
+WebSocket Client
+----------------
```

Comment:
Maybe have a more descriptive title for this section

```suggestion
WebSocket Non Web Browser Client
--------------------------------
```

Reply:
The longer section titles just looked ugly when I tried them, but I added some verbiage to make it clearer what kinds of client and server we are talking about